Mathlib/CategoryTheory/Category/Cat/Adjunction.lean

```diff
@@ -51,4 +49,48 @@ def typeToCatObjectsAdj : typeToCat ⊣ Cat.objects where
           obtain rfl := Discrete.eq_of_hom f
           aesop_cat ) }
 
+
+/-- The connected components functor -/
+def connectedComponents.{v} : Cat.{v, u} ⥤ Type u where
+  obj C := ConnectedComponents C
+  map F :=
+    Quotient.lift (Quotient.mk (Zigzag.setoid _) ∘ F.obj)
+      (fun _ _ ↦ Quot.sound ∘ zigzag_obj_of_zigzag F)
+  map_id _ := funext fun x ↦ (Quotient.exists_rep x).elim (fun _ h ↦ by simp [<- h]; rfl)
+  map_comp _ _ := funext fun x ↦ (Quotient.exists_rep x).elim (fun _ h => by simp [<- h])
+
+
+private def fnToFctr  {X C} (fct : connectedComponents.obj C ⟶ X) : (C ⥤ typeToCat.obj X) where
```

Comment:
This shouldn't be a private def, it should look something like this and be in a different file:

```lean
@[simps]
def ConnectedComponents.functorToDiscrete {X C : Type*} [Category C] (f : ConnectedComponents C → X) : 
    C ⥤ Discrete X where
  obj :=  Discrete.mk ∘ f ∘ Quotient.mk (Zigzag.setoid _)
  map :=  Discrete.eqToHom ∘ congrArg f ∘ Quotient.sound ∘ Zigzag.of_hom
```

In particular, it shouldn't mention `Cat` or the functors `connectedComponents` and `typeToCat`.

Reply:
makes sense